_posts/2023-09-21-hybrid-search.md

```diff
@@ -0,0 +1,390 @@
+
+It’s important to remember that datasets have different parameters and for some datasets hybrid search may not consistently improve results. Additionally, experiments with different parameters&mdash;for instance, setting higher values of `k` or use different `space type` for neural search&mdash;may produce better results. 
+
+We’ve seen that some of the conclusions may be applied to most datasets:
+
+* For semantic search, hybrid query with normalization produces better results compared to neural search or text search alone.
+* The combination of `min-max` score normalization and `arithmetic_mean` score combination achieves the best results, compared to other techniques.
+* In most cases, increasing the value of `k` in k-NN data type leads to better results up to a certain point, but after that there is no increase in relevance. At the same time, high values of `k` increase search latency, so from our observations, it’s better to choose the value of `k` between 100 and 200. 
+* We have seen best results when `innerproduct` is specified as a space type for k-NN vector fields. This may be because our models were trained using the inner product similarity function.
+* An increase in search relevance comes with a cost of a 6&ndash;8% increase in latency, which should be acceptable in most cases.
+
+In general, as our experiments demonstrated, hybrid search produces results that are very close to the ones described by the science team, so [all their conclusions](https://opensearch.org/blog/semantic-science-benchmarks/#section-5-strengths-and-limitations) are applicable to hybrid search. 
+
+## Next steps
+
+We have identified several areas of improvement for hybrid search, and we’re planning to work on them for the future OpenSearch releases. In the short term, a good starting point is to improve performance by running individual queries of the main hybrid query in parallel instead of sequentially. This should significantly improve latency, especially when all inner queries have similar running times. We are considering the following improvements for the future releases:
+
+* Executing individual queries in parallel.
+* Adding more configuration options and parameters to the normalization processor to allow more control over combined results. For instance, we can add the ability to specify a minimal score for documents to be returned in the results, which will avoid returning non-competitive hits. 
+* Supporting results pagination
+* Supporting filters in the hybrid query clause. It’s possible to define a filter for each inner query individually, but it’s not optimal if a filter condition is the same for all inner queries.
+* Adding more benchmark results for larger datasets so we can provide recommendations for using hybrid search in various configurations. 
+
```

Comment:
we have github issues around all of them. @martin-gaievski please provide those issues here

Reply:
sure, here are the issues I opened for future work:

- https://github.com/opensearch-project/neural-search/issues/279
- https://github.com/opensearch-project/neural-search/issues/280
- https://github.com/opensearch-project/neural-search/issues/281
- https://github.com/opensearch-project/neural-search/issues/282
- https://github.com/opensearch-project/neural-search/issues/299